### DISCUSSION TITLE
How to adjust the talk port?

### DISCUSSION THREAD

**Discussion by ghost** [nextcloud/all-in-one#986] [Questions] [answered]
I can't find any documentation on this.
Can anyone help me?

**Comment by Zoey2936** [comment]
If you use docker-compose you can use this option here https://github.com/nextcloud/all-in-one/blob/c4e7a0ea0ba4c7a8994d22075df4f2d87a28bbe0/docker-compose.yml#L22
And if you use the docker run command, just add to the start command `-e "TALK_PORT=3478"` (but change 3478 to your wanted port.

**Reply by ghost** [reply]
I have been trying for a while [Multiple AIO instances](https://github.com/nextcloud/all-in-one/blob/main/multiple-instances.md) To try, unfortunately I do not succeed.

I have proceeded as follows:
1. created new users with sudo adduser [user].
2. made these users superusers with sudo usermod -aG sudo [user]
3. then I ran the following with [user1]:
# For x64 CPUs:
sudo docker run -it \
--name nextcloud-aio-mastercontainer \
--restart always \
-p 8081:8080 \
-e APACHE_PORT=11001 \
-e TALK_PORT=3479
-e DOCKER_SOCKET_PATH=$XDG_RUNTIME_DIR/docker.sock
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume $XDG_RUNTIME_DIR/docker.sock:/var/run/docker.sock:ro
nextcloud/all-in-one:latest

so i tried to adjust the port unfortunately i got the following message:
"docker run requires at least 1 argument.
See 'docker run --help'.

**Reply by Zoey2936** [reply]
you miss the ` \` a the end of each line

**Reply by ghost** [reply]
sudo docker run -it \
> --name nextcloud-aio-mastercontainer \
> --restart always \
> -p 8081:8080 \
> -e APACHE_PORT=11001 \
> -e TALK_PORT=3479 \
> -e APACHE_IP_BINDING=127.0.0.1 \
> -e DOCKER_SOCKET_PATH=$XDG_RUNTIME_DIR/docker.sock \
> --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
> --volume $XDG_RUNTIME_DIR/docker.sock:/var/run/docker.sock:ro \
> nextcloud/all-in-one:latest

Thank you very much I do not know how this could happen. But now I get the following error message.
sudo: docker: command not found

**Reply by Zoey2936** [reply]
Did you add the \ to every line, or only the first?
```
sudo docker run -it \
    --name nextcloud-aio-mastercontainer \
    --restart always \
    -p 8081:8080 \
    -e APACHE_PORT=11001 \
    -e TALK_PORT=3479 \
    -e APACHE_IP_BINDING=127.0.0.1 \
    -e DOCKER_SOCKET_PATH=$XDG_RUNTIME_DIR/docker.sock \
    --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
    --volume $XDG_RUNTIME_DIR/docker.sock:/var/run/docker.sock:ro \
    nextcloud/all-in-one:latest
```

**Reply by ghost** [reply]
To every line.

**Reply by Zoey2936** [reply]
hmm what says the output of
`sudo command -v docker`

**Reply by ghost** [reply]
sudo: command: command not found

**Reply by Zoey2936** [reply]
Hmm, is this the output of every command?

**Reply by ghost** [reply]
> Hmm, is this the output of every command?

it seems so :(

**Comment by szaimen** [comment, marked as answer]
See https://github.com/nextcloud/all-in-one#how-to-adjust-the-talk-port
